### PR TITLE
Made NamespaceManager.Run idempotent and added graceful shutdown

### PR DESCRIPTION
**Pull request by markturansky**
@derekwaynecarr I patterned my recent controller after this one, but added a graceful shutdown that I think makes sense for controllers.  Also made the Run method idempotent.

### CODE REVIEW

File: pkg/namespace/namespace_controller_test.go [hunk 1]
```diff
@@ -131,3 +133,25 @@ func TestSyncNamespaceThatIsActive(t *testing.T) {
...
+	nsMgr := NewNamespaceManager(client, 1*time.Second)
+
+	if nsMgr.StopEverything != nil {
+		t.Errorf("Non-running manager should not have a stop channel.  Got %v", nsMgr.StopEverything)
+	}
+
+	nsMgr.Run()
+
+	if nsMgr.StopEverything == nil {
+		t.Errorf("Runnig manager should have a stop channel.  Got nil")
```
Comment (derekwaynecarr) [inline pkg/namespace/namespace_controller_test.go]: Running

**Comment by derekwaynecarr** [comment]
Minor typo nit, otherwise LGTM

**Comment by markturansky** [comment]
Nit picked, @derekwaynecarr.  Travis is green.

**Comment by quinton-hoole** [comment]
LGTM.  Mark, are there and e2e tests that cover this, and have you run them?

**Comment by markturansky** [comment]
@quinton-hoole Whatever tests that run and cover the namespace controller would test running the controller.  The graceful shutdown is unit tested but not used yet in code.  The kube-controller-manager doesn't (yet!) gracefully shutdown controllers but I intend for it to do so in #6520.

**Comment by derekwaynecarr** [comment]
LGTM

**Pull request merged**